javalink/ref.py

```diff
@@ -238,19 +252,21 @@ def __str__(self):
 
 def initialize_package_list(app):
     env = app.env
-    if hasattr(env, 'javalink_packages'):
+    if hasattr(env, 'javalink_packages') and hasattr(env, 'javalink_packages_versions'):
         return
 
     app.verbose('[javalink] initializing package list...')
     env.javalink_packages = {}
+    env.javalink_packages_versions = {}
 
-    for url, base in [parse_docroot(env.srcdir, r) for r in app.config.javalink_docroots]:
+    for javadoc_version, url, base in [parse_docroot(env.srcdir, javadoc_version, r) for javadoc_version, r in app.config.javalink_docroots]:
```

Comment:
I think this breaks the 3-tuple (version, root, base) case - `a, b in list` will throw an exception when the list contains a tuple with more than 2 elements.